packages/superset-ui-connection/types/external.d.ts

```diff
@@ -1 +1,8 @@
 declare module 'fetch-mock';
+
+declare module 'json-bigint' {
```

Comment:
@xtinec do you have any thoughts on this type of declaration? it looks okay to me, not sure if we can be more specific than `any`.

Reply:
I think we could do `BigNumber | string` here instead of `any`. 

Looks like `json-bigint` doesn't have a type definition on DefinitelyTyped but we can add one for it pretty easily (it only has two exported methods and we already wrote the type for one of them here 🙂). Then, we could install our new type definition as a dependency. @soboko would you be interested in doing it?


P.S. Unrelated but we could also add the type def for `fetch-mock` which is already available on `DefinitelyTyped` @williaster

Reply:
let us know what you think about this @soboko, this could be done in a separate PR if you want to get this merged.

Reply:
I used `any` as the return type for consistency with the return type of `JSON.parse`. I did just add the second method and stole some comments from `lib.es5.d.ts`... 

I agree with adding a type def for `json-bigint` to DefinitelyTyped is the right thing to do and I can certainly do that. Looks like it could take up to a week for that PR to be accepted. Perhaps we could still merge this PR for the time being? There's a PR in `incubator-superset` that's dependent on this one.

I also added the type def for `fetch-mock` in my latest commit.

Reply:
So I looked into creating a proper type def file for DefinitelyTyped, and there's a bit of a complication.

Specifically, the [API for json-bigint](https://github.com/sidorares/json-bigint) looks like:

```js
module.exports = function(options) {
    return  {
        parse: json_parse(options),
        stringify: json_stringify
    }
};
//create the default method members with no options applied for backwards compatibility
module.exports.parse = json_parse();
module.exports.stringify = json_stringify;
```

The idea is that you can construct a parser with non-default options using something like:
```js
var JSONstrict = require('json-bigint')({"strict": true});
```

AFAICT there's no clean way to reconcile the json-bigint API with ES6/TypeScript-style imports and type definition files.

I've [submitted a PR](https://github.com/sidorares/json-bigint/pull/27) that would allow using something like the following to construct a "special" JSONbig parser:

```js
const JSONbigspecial = JSONbig.newInstance({ storeAsString: true, strict: true });
```

If that PR is merged, then I can submit a decent type declaration file to DefinitelyTyped and problem solved.